File: Gestion_Evenement2/src/main/java/edu/esprit/entities/Messagerie.java
package edu.esprit.entities;

import java.sql.Date;
import java.sql.Timestamp; // Import Timestamp for handling datetime
import java.util.Objects;

public class Messagerie {
    private int id;
    private String nom;
    private Date date;
    private String message;
    private int idu; // Assuming this corresponds to the "idu" field in the database

    public Messagerie(int id, String nom, String message) {
        this.id = id;
        this.nom = nom;
        this.message = message;
    }

    // Assuming you want to initialize the date with the current date and time
    public Messagerie(String nom, String message, int idu) {
        this.nom = nom;

        this.message = message;
        this.idu = idu;
    }
   

    public Messagerie(int id, String nom, String message, int idu) {
        this.id = id;
        this.nom = nom;
        long currentTimeMillis = System.currentTimeMillis();

        this.date = new Date(currentTimeMillis);
        this.message = message;
        this.idu = idu;
    }

    public Messagerie(int id, String nom, Date date, String message) {
        this.id = id;
        this.nom = nom;
        this.date = date;
        this.message = message;
    }



    public Messagerie(String nom, Date date, String message, int idu) {
        this.nom = nom;
        // Assuming date is a String, you may need to convert it to a Date or Timestamp
        long currentTimeMillis = System.currentTimeMillis();

        this.date = new Date(currentTimeMillis);
        this.message = message;
        this.idu = idu;
    }

    public Messagerie(int id, String nom, Timestamp date, String message) {
    }


    // Getter and Setter methods...

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public int getIdu() {
        return idu;
    }

    public void setIdu(int idu) {
        this.idu = idu;
    }

    @Override
    public String toString() {
        return "Messagerie{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", date=" + date +
                ", message='" + message + '\'' +
                ", idu=" + idu +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Messagerie that = (Messagerie) o;
        return id == that.id;
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: projettest-main2/src/Entity/Messagerie.php
<?php

namespace App\Entity;

use App\Repository\MessagerieRepository;
use DateTime;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints\Date;

#[ORM\Entity(repositoryClass: MessagerieRepository::class)]



class Messagerie
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null ;

    #[ORM\Column(length: 255)]
    private ?string $nom;

    #[ORM\Column]

    private ?Date $date;

    #[ORM\Column(length: 255)]
    private ?string $message;

    #[ORM\ManyToOne(inversedBy: 'Messageries')]
    #[ORM\Column]
    private ?Utilisateurs $idu = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): static
    {
        $this->nom = $nom;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): static
    {
        $this->date = $date;

        return $this;
    }

    public function getMessage(): ?string
    {
        return $this->message;
    }

    public function setMessage(string $message): static
    {
        $this->message = $message;

        return $this;
    }

    public function getIdu(): ?Utilisateurs
    {
        return $this->idu;
    }

    public function setIdu(?Utilisateurs $idu): static
    {
        $this->idu = $idu;

        return $this;
    }
    public function __toString()
    {
        return $this->getId();
    }

}